runtime/src/main/java/com/dylibso/chicory/runtime/Memory.java

```diff
@@ -40,6 +41,272 @@ static int bytes(int pages) {
 
     int notify(int address, int maxThreads);
 
+    default int atomicLoadInt(int addr) {
```

Comment:
@andreaTP I went for "load" cause I was looking at the spec, but in retrospect I think "read" is more consistent with this repo. Would you prefer `atomic(Read/Write)<type>`?

I also understand the `<type>` as using... ok I think I see the convention now, and I didn't follow it. All write methods can use Java type names and corresponding type i.e. `Byte`, `Short`, `Int` and `Long` it's unambiguous. All read methods should specify `<type>` as `<U|I><width>` and always return `long` with the correct sign extension.

Is this correct? I guess there is one offending example in `writeI32(...)`...